articles/documentation-docfx.md

```diff
@@ -0,0 +1,300 @@
+Markdown docs location (docs): articles
+Enable site search? [y/n] (y): y
+Enable PDF? [y/n] (y): n
+```
+
+Congrats! You now have a minimal `docfx` website. Run the following command to generate a local preview of the website, and verify that API documentation for your package has been generated successfully:
+
+```powershell
+dotnet docfx docfx.json --serve
+```
+
+### Docfx folder organization
+
+Before proceeding further into configuring the `docfx` website, the `docs` folder should be organised according to this layout for effective long term maintenance. 
+
+```
+.
+├── api/                     # Automatically generated by docfx
+├── apidoc/                  # Place individual operator articles here (see style guide)
+├── articles/                # Place "Manual" section articles here (see style guide) 
+├── bonsai/                  # Docfx-tools submodule (see note below)
+├── images/                  # Place images like screenshots here
+├── template/                # Custom CSS template (see note below)
+│   └── public/
+│       ├── main.css
+│       └── main.js
+├── tutorial/                # Place "Tutorials" section articles here (see style guide)   
+├── workflow/                # Place .bonsai workflow files here for image export
+├── build.ps1                # Bonsai workflow image export script (see note below)
+├── logo.svg                 # Website logo
+└── favicon.ico              # Website bookmark logo
+...
+```
+
+- `bonsai/` - this folder is where we place a submodule that we have created called [docfx-tools](https://github.com/bonsai-rx/docfx-tools). This submodule contains scripts for automating image export from `Bonsai` workflows (if you want to include them in your article) and adds a CSS template for `Bonsai` workflow containers (these containers allow users to easily copy workflows into the `Bonsai` editor). To add it, run this command in the `docs` folder:
+
+```powershell
+git submodule add https://github.com/bonsai-rx/docfx-tools bonsai
+```
+If you cloned the repository and the `bonsai` folder does not contain any files:
+
+```powershell
+git submodule update --init
+```
+
+To update the submodule to the latest commit:
+```powershell
+git submodule update --remote
+```
+
+- `template/` - this custom CSS template patches in docfx-tools and adds a github link to the top right of the navigation bar. Download this folder from the [docfx-assets](https://github.com/bonsai-rx/docfx-assets) repository and amend the github link in `main.css` to your repository.
+
+- `build.ps1` - this custom script exports Bonsai workflow images using `docfx-tools` and then calls the docfx build process. Download the file from the [docfx-assets](https://github.com/bonsai-rx/docfx-assets) repository and amend the line that specifies the package name and source location.
+
+- `logo.svg` and `favicon.ico` - for official Bonsai packages these can be downloaded from the [docfx-assets](https://github.com/bonsai-rx/docfx-assets) repository.
```

Comment:
I did start creating some python scripts to automate some of the docfx setup (see https://github.com/bonsai-rx/docfx-tools/pull/11 for a script to configure docfx.json) but now that you mention it, Bonsai.Templates might be a better place to have a documentation template. I guess this will also solve the issue of where to put the assets. Is there a GitHub repository for it to raise an issue so we can keep track of it?